services/dataService.js

```diff
@@ -0,0 +1,62 @@
+const productMap = {};
+const categoryMap = {};
+const combinedProductMap = {};
```

Comment:
See above.

Reply:
Because I;m using a prototype function `Object.entries()`.  I am not allowed to declare the variable as null(because then I cannot use `Object.entries()`. Also, I've made it `const` because I'm only changing the properties of the object and not the object itself. So it still works as expected